app/src/main/java/com/example/riden/Adapters/RideAdapter.java

```diff
@@ -177,85 +179,82 @@ protected FilterResults performFiltering(CharSequence constraint) {
-                                        Double lat = location.getDouble("lat");
-                                        Double lng = location.getDouble("lng");
-
-//                                        searchDestination = new MarkerOptions().position(new LatLng(lat, lng)).title("searchDestination");
-
-                                        for(Ride ride: allRides) {
-//                                            rideDestination = new MarkerOptions().position(new LatLng(ride.getDestinationLat(), ride.getDestinationLong())).title("rideDestination");
-                                            float[] distanceBetween = new float[1];
-                                            Location.distanceBetween(lat, lng, ride.getDestinationLat(), ride.getDestinationLong(), distanceBetween);
-                                            float miles = distanceBetween[0]/1609;
-                                            if(miles < 20) {
-                                                rides.add(ride);
-                                            }
-                                        }
-                                    }
-                                    Log.i("a_new_response", jsonResponse.toString());
-                                } catch (JSONException e) {
-                                    e.printStackTrace();
+    public void callToGoogleMapsSearch(String query) {
+        OkHttpClient client = new OkHttpClient().newBuilder()
+                .build();
+
+        String urlBeforeQuery = "https://maps.googleapis.com/maps/api/place/textsearch/json?query=";
+
+        Request request = new Request.Builder()
+                .addHeader("Accept", "application/json")
+                .url(urlBeforeQuery + query + "&key=" + context.getString(R.string.google_maps_api_key))
+                .method("GET", null)
+                .build();
+
+        client.newCall(request).enqueue(new Callback() {
+            @Override
+            public void onFailure(Call call, IOException e) {
+                e.printStackTrace();
+            }
+
+            @Override
+            public void onResponse(Call call, final Response response) throws IOException {
+                if (!response.isSuccessful()) {
+                    throw new IOException("Unexpected code " + response);
+                }
+                else {
+                    try {
+                        JSONObject jsonResponse = new JSONObject(response.body().string());
+                        JSONArray results = jsonResponse.getJSONArray("results");
+
+                        if (results != null) {
+                            rides.clear();
+                            JSONObject geometry = results.getJSONObject(0).getJSONObject("geometry");
+                            JSONObject location = geometry.getJSONObject("location");
+                            Double lat = location.getDouble("lat");
```

Comment:
Can we store the lat and long for each location when creating the ride ? That way we don't have to fetch it again for each location when searching. That can greatly improve search perf